src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -143,8 +144,18 @@ public static void AssertValidValue(ISchema schema, IGraphType type, object inpu
 
             if (type is ScalarGraphType scalar)
             {
-                if (ValueFromScalar(scalar, input) == null)
-                    throw new InvalidValueException(fieldName, "Invalid Scalar value for input field.");
+                // validate value can be parsed correctly
+
+                if (input is IValue value)
+                {
+                    if (scalar.ParseLiteral(value) == null)
+                        throw new InvalidValueException(fieldName, $"Unable to convert '{value.Value}' to {type.Name}");
```

Comment:
Lets rename `fieldName` to `variableName` in ` public static void AssertValidValue(ISchema schema, IGraphType type, object input, string fieldName)` and also in `InvalidValueException` ctor. It's not really a variable name, because the check can happen on array elements and on nested properties, but nevertheless it's a better name than `fieldName`, because this method is made for input values. So I even think it would be better to rename `AssertValidValue` to `AssertValidVariableValue`.

Reply:
no problem; hang on a sec

Reply:
yeah me too.  only during debugging did I realize it was only for variables